src/frontend/servers/y-provider/__tests__/server.test.ts

```diff
@@ -0,0 +1,230 @@
+import {
+  HocuspocusProvider,
+  HocuspocusProviderWebsocket,
+} from '@hocuspocus/provider';
+import request from 'supertest';
+import WebSocket from 'ws';
+
+const port = 5555;
+const portWS = 6666;
+const origin = 'http://localhost:3000';
+
+jest.mock('../src/env', () => {
+  return {
+    PORT: port,
+    COLLABORATION_SERVER_ORIGIN: origin,
+    COLLABORATION_SERVER_SECRET: 'test-secret-api-key',
+  };
+});
+
+console.error = jest.fn();
+
+import { hocuspocusServer, initServer } from '../src/server'; // Adjust the path to your server file
+
+const { app, server } = initServer();
+
+describe('Server Tests', () => {
+  beforeAll(async () => {
+    await new Promise<void>((resolve) => {
+      if (server.listening) {
+        resolve();
+      }
+    });
+
+    await new Promise<void>((resolve) => {
```

Comment:
you can just do `await hocuspocusServer
        .configure({
          port: portWS,
        })
        .listen()`

Personally, I try to avoid `.then` and `.catch` as much as possible and prefer `async` / `await`

Reply:
https://github.com/numerique-gouv/impress/pull/472/commits/e7565f06795b35036580aa9ace4e3d0d81c3795b#diff-28f2602f409539020f08398a838eea589448bd58ccf86f9670f9c06f1fbeeef4R28-R30

I added this helper, to reduce promise hell:
https://github.com/numerique-gouv/impress/pull/472/commits/e7565f06795b35036580aa9ace4e3d0d81c3795b#diff-7b4e113632ac6a5668894882898d4c59b8260390aa9c61ebe6962395e4b5c94e